tests/SprykerEcoTest/Zed/Payone/Business/Distributor/OrderPriceDistributorTest.php

```diff
@@ -0,0 +1,344 @@
+            (int)$itemsPrices[ItemTransfer::UNIT_PRICE_TO_PAY_AGGREGATION] + $expensesPrices,
+            'Items unit to aggregation and expenses total price should be equals to the expected price.'
+        );
+
+        $this->assertSame(
+            $expectedTotalPrice,
+            (int)$itemsPrices[ItemTransfer::UNIT_GROSS_PRICE] + $expensesPrices,
+            'Items unit gross and expenses total price should be equals to the expected price.'
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return int[]
+     */
+    protected function getOrderItemsPrices(OrderTransfer $orderTransfer): array
+    {
+        $prices = [
+            ItemTransfer::SUM_PRICE_TO_PAY_AGGREGATION => 0,
+            ItemTransfer::UNIT_PRICE_TO_PAY_AGGREGATION => 0,
+            ItemTransfer::UNIT_GROSS_PRICE => 0,
+        ];
+
+        foreach ($orderTransfer->getItems() as $itemTransfer) {
+            $prices[ItemTransfer::SUM_PRICE_TO_PAY_AGGREGATION] += $itemTransfer->getSumPriceToPayAggregation();
+            $prices[ItemTransfer::UNIT_PRICE_TO_PAY_AGGREGATION] += $itemTransfer->getUnitPriceToPayAggregation() * $itemTransfer->getQuantity();
+            $prices[ItemTransfer::UNIT_GROSS_PRICE] += $itemTransfer->getUnitGrossPrice() * $itemTransfer->getQuantity();
+        }
+
+        return $prices;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return int
+     */
+    protected function getOrderExpensesTotalPrice(OrderTransfer $orderTransfer): int
+    {
+        $totalPrice = 0;
+        foreach ($orderTransfer->getExpenses() as $expenseTransfer) {
+            $totalPrice += $expenseTransfer->getSumGrossPrice();
+        }
+
+        return $totalPrice;
+    }
+
+    /**
+     * @param mixed[] $seedData
+     *
+     * @return \Generated\Shared\Transfer\ItemTransfer
+     */
+    protected function buildItemTransfer(array $seedData = []): ItemTransfer
+    {
+        return (new ItemBuilder($seedData))->build();
+    }
+
+    /**
+     * @param mixed[] $seedData
+     *
+     * @return \Generated\Shared\Transfer\ExpenseTransfer
+     */
+    protected function buildExpenseTransfer(array $seedData = []): ExpenseTransfer
+    {
+        return (new ExpenseBuilder($seedData))->build();
+    }
+
+    /**
+     * @param mixed[] $seedData
+     *
+     * @return \Generated\Shared\Transfer\PaymentTransfer
+     */
+    protected function buildPaymentTransfer(array $seedData = []): PaymentTransfer
+    {
+        return (new PaymentBuilder($seedData))->build();
+    }
+
+    /**
+     * @param mixed[] $seedData
+     *
+     * @return \Generated\Shared\Transfer\TotalsTransfer
+     */
+    protected function buildTotalsTransfer(array $seedData = []): TotalsTransfer
+    {
+        return (new TotalsBuilder($seedData))->build();
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideOrdersData(): array
```

Comment:
```suggestion
    public function distributePricesShouldReturnOrderTransferWithDistributedPricesDataProvider(): array
```

Reply:
It's too descriptive, we can use this method in any test for sure. Let it be `orderPriceForDistributionDataProvider`.